catalog/ui/catalog-ui-search/src/main/webapp/component/session-timeout-modal/session-timeout.controller.js

```diff
@@ -0,0 +1,96 @@
+ **/
+/* jshint browser: true */
+/* global define */
+define(['jquery',
+        'underscore',
+        'marionette',
+        'js/wreqr',
+        './lightbox/lightbox.view',
+        './session-timeout.view'
+    ], function ($, _, Marionette, wreqr, BlockingLightbox, SessionTimeoutModal) {
+        'use strict';
+        var SessionTimeoutController;
+
+        var _this = null;
+        var idleNoticeDuration = 60000;
+        var idleTimeoutThreshold = 900000;  // Length of inactivity that will trigger user timeout
+                                            // See STIG V-69243
+
+        SessionTimeoutController = Marionette.Object.extend({
+            initialize: function () {
+                _this = this;
+                _this.initTimer();
+
+                wreqr.vent.on("sessionRenewed", _this.initTimer);
+            },
+
+            initTimer: function () {
+                var timer = null;
+                var sessionRenewDate = 0;
+                var idleTimeoutDate = idleTimeoutThreshold - idleNoticeDuration + Date.now();
+                var modalShown = false;
+
+                wreqr.vent.on("continueClicked", function () {
+                    modalShown = false;
+                });
+
+                function setSessionRenewDate() {
+                    $.get("/services/session/expiry", function (res) {
+                        var msUntilTimeout = parseInt(res);
+                        if (msUntilTimeout === 0)
+                            clearTimeout(timer);
+
+                        var msUntilAutoRenew = Math.max(msUntilTimeout * 0.7, msUntilTimeout - 60000);
+                        sessionRenewDate = Date.now() + msUntilAutoRenew;
+                    });
+                }
+
+                function onInterval() {
+                    var now = Date.now();
+
+                    if (now >= sessionRenewDate) {
+                        $.get("/services/session/renew", function () {
+                            console.log("session renewed");
+                            setSessionRenewDate();
+                        });
+                    }
+
+                    if (now >= idleTimeoutDate && !modalShown) {
+                        $(document).off('keydown mousedown scroll');
+                        var blockingLightbox = BlockingLightbox.generateNewLightbox();
+
+                        blockingLightbox.model.updateTitle("Session Expiring");
+                        blockingLightbox.model.open();
+                        blockingLightbox.lightboxContent.show(new SessionTimeoutModal({
+                            time: idleNoticeDuration
+                        }));
+                        modalShown = true;
+                    }
+                }
+
+                function onUserActivity() {
+                    idleTimeoutDate = idleTimeoutThreshold - idleNoticeDuration + Date.now();
+                }
+
+                setSessionRenewDate();
+                timer = setTimeout(onInterval, 1000);
```

Comment:
The callback passed to `setTimeout` is called `onInterval`. Did we mean to have `setInterval`?

Reply:
No, `onInterval` is correct. This code is basically an `setInterval` implementation, but it uses `setTimeout` so that the timer inaccuracies can be corrected at each interval - hence the `onInterval` callback.

Reply:
I can't seem to find where `onInterval` is being run after the initial timeout. How are we achieving the goal of checking on an interval?